srtcore/core.cpp

```diff
@@ -4852,11 +4852,11 @@ void *CUDT::tsbpd(void *param)
             HLOGC(tslog.Debug, log << self->CONID() << "tsbpd: no data, scheduling wakeup at ack");
             self->m_bTsbPdAckWakeup = true;
             THREAD_PAUSED();
-            pthread_cond_wait(&self->m_RcvTsbPdCond, &self->m_RecvLock);
+            pthread_cond_wait(&self->m_RcvTsbPdCond, &(self->m_RecvLock.ref()));
             THREAD_RESUMED();
         }
     }
-    CGuard::leaveCS(self->m_RecvLock);
+    recv_lock.unlock();
```

Comment:
Unnecessary. Can be unlocked in destructor here.

Reply:
I think this unlock is done to have the following section out of the guard:
```
THREAD_EXIT();
HLOGC(tslog.Debug, log << self->CONID() << "tsbpd: EXITING");
```

Reply:
So, no. This `THREAD_EXIT()` is an entry for Makito thread tracer stuff, it should actually be at the very end of this function. The logging instruction is also alright to be under or outside the CS.

Reply:
You are probably right on the usage.
Still, I would like to retain the previous behavior for convenience. And previously the mutex was being explicitly unlocked.

Reply:
Yes, but it was intended to be unlocked at the end of the function. Previously, before I added the logging instruction, it was really at the end, only followed by `THREAD_EXIT()`. I simply added this log by jumping to the end of this function, ignoring whatever was there before. I confirm officially: it wasn't my intention to have the mutex unlocked specifically before the logging instruction. Also, the original author of this code simply didn't use `CGuard`.